From 3eaedda4caf678b5cc64f3595819170994ac44c4 Mon Sep 17 00:00:00 2001
From: "sheng.liu" <sheng.liu@amlogic.com>
Date: Fri, 17 Sep 2021 10:42:15 +0800
Subject: [PATCH] gst-plugins-good-1.16.3: donot treat ac4 chunks as sample
 [1/1]

PD#SWPL-58849

Problem:
treat audio ac4 codec chunks as samples
which will make sample size too large to decode

Solution:
Do not treat audio ac4 codec chunks as samples

Verify:
AP222

Change-Id: Ie420bcfce298c0a4934bf0d94909dc4af5e867b9
---
 gst/isomp4/qtdemux.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gst/isomp4/qtdemux.c b/gst/isomp4/qtdemux.c
index e38f6dd..c0af7fb 100644
--- a/gst/isomp4/qtdemux.c
+++ b/gst/isomp4/qtdemux.c
@@ -15314,6 +15314,7 @@ qtdemux_audio_caps (GstQTDemux * qtdemux, QtDemuxStream * stream,
       _codec ("AC-4 audio");
       caps = gst_caps_new_simple ("audio/x-ac4",
           "framed", G_TYPE_BOOLEAN, TRUE, NULL);
+      entry->sampled = TRUE;
       break;
     case GST_MAKE_FOURCC ('d', 't', 's', 'c'):
     case GST_MAKE_FOURCC ('D', 'T', 'S', ' '):
-- 
2.32.0

